uint8 id 
float32[] rvec
float32[] tvec


